src/options.c

```diff
@@ -369,30 +414,73 @@ JOHN_USAGE_REGEX \
+"--config=FILE              use FILE instead of john.conf or john.ini\n" \
+"--log-stderr               log to screen instead of file\n"             \
+"--verbosity=N              change verbosity (1-%u or %u for debug, default %u)\n" \
+"--no-log                   disables creation and writing to john.log file\n"      \
+"--bare-always-valid=Y      treat bare hashes as valid (Y/N)\n" \
+"--progress-every=N         emit a status line every N seconds\n" \
+"--crack-status             emit a status line whenever a password is cracked\n" \
+"--keep-guessing            try finding plaintext collisions\n" \
+"--no-keep-guessing         turn off any default \"keep guessing\" for used format\n" \
+"--max-candidates=[-]N      gracefully exit after this many candidates tried.\n" \
+"                           (if negative, reset count on each crack)\n" \
+"--max-run-time=[-]N        gracefully exit after this many seconds (if negative,\n" \
+"                           reset timer on each crack)\n" \
+"--regen-lost-salts=N       brute force unknown salts (see doc/OPTIONS)\n" \
 JOHN_USAGE_FORK \
 "--pot=NAME                 pot file to use\n" \
-"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n"
+"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n" \
+"--input-encoding=NAME      input encoding (alias for --encoding)\n" \
+"--internal-codepage=NAME   codepage used in rules/masks (see doc/ENCODINGS)\n" \
+"--target-encoding=NAME     output encoding (used by format, see doc/ENCODINGS)\n" \
+"--tune=HOW                 tuning options (auto/report/N)\n"
 
 #define JOHN_USAGE_FORMAT \
-"--format=NAME              force hash of type NAME. The supported formats can\n" \
-"                           be seen with --list=formats and --list=subformats\n\n"
+"--format=[NAME|CLASS][,..] force hash of type NAME. The supported formats can\n" \
+"                           be seen with --list=formats and --list=subformats. Valid\n" \
+"                           classes: dynamic, cpu, opencl, ztex, mask, omp, all,\n" \
+"                           enabled, disabled.\n"
 
 #if defined(HAVE_OPENCL)
 #define JOHN_USAGE_GPU \
-"--devices=N[,..]           set OpenCL device(s) (see --list=opencl-devices)\n"
+"--devices=N[,..]           set OpenCL device(s) (see --list=opencl-devices)\n" \
+"\nOpenCL options:\n" \
+"--force-scalar            force scalar mode\n" \
+"--force-vector-width=N    force vector width N\n" \
+"--lws=N                   force local worksize N\n" \
+"--gws=N                   force global worksize N\n\n"
 #define JOHN_USAGE_ZTEX \
 "                           or set ZTEX device(s) by its(their) serial number(s)\n"
 #elif defined(HAVE_ZTEX)
 #define JOHN_USAGE_ZTEX \
 "--devices=N[,..]           set ZTEX device(s) by its(their) serial number(s)\n"
 #endif
 
+static void print_banner(char *name)
+{
+	if (!john_main_process)
+		exit(0);
```

Comment:
469-470 breaks MPI: You must not exit unless main node will also exit!

Simply doing this would work fine though (not sure if it's worth its own function at all though?
```c
	if (!john_main_process)
	    exit(0);
```